airflow/models/dagrun.py

```diff
@@ -346,7 +331,38 @@ def update_state(self, session=None):
         session.merge(self)
         session.commit()
 
-        return self.state
+        return ready_tis
+
+    @provide_session
+    def get_ready_tis(self, session, scheduleable_tasks, finished_tasks):
```

Comment:
This feels more like a private method, especially since it takes two list of tasks - I think we should make this "private" `_get_ready_tis` and remove the `@provide_session` decorator too so that we require to pass in and use the same DB session.

For consistency with the rest of the code please also make session the last argument.